File: src/sp/it/pl/gui/itemnode/textfield/FontItemNode.java
package sp.it.pl.gui.itemnode.textfield;

import javafx.scene.text.Font;
import sp.it.pl.gui.objects.picker.FontSelectorDialog;
import sp.it.pl.util.parsing.Parser;

public class FontItemNode extends TextFieldItemNode<Font> {

	public FontItemNode() {
		super(Parser.DEFAULT.toConverterOf(Font.class));
	}

	@Override
	void onDialogAction() {
		new FontSelectorDialog(getValue(), this::setValue).show(this);
	}
}

File: src/sp/it/pl/gui/itemnode/textfield/FontItemNode.kt
package sp.it.pl.gui.itemnode.textfield

import javafx.scene.text.Font
import sp.it.pl.gui.objects.picker.FontSelectorDialog
import sp.it.pl.main.AppUtil.APP

/** Text field for [Font] with a picker. */
class FontItemNode: TextFieldItemNode<Font>({ APP.converter.general.toS(it) }) {

    override fun onDialogAction() {
        FontSelectorDialog(value, { value = it }).showInCenterOf(this)
    }

}